lib/puppet/provider/cpan/default.rb

```diff
@@ -31,12 +31,14 @@ def create
       ll = "-Mlocal::lib=#{resource[:local_lib]}"
     end
 
+    umask = "umask #{resource[:umask]};" if resource[:umask]
```

Comment:
is there a default value for umask ? if not, wouldn't this generate an error ?

Reply:
We could use 022 as the default but it could be argued that would make it a breaking change?  Maybe someone is relying on the current behaviour??  Probably not.

I think the code is fine though.  It would be functionally identical to write the single line as...
```ruby
if resource[:umask]
  umask = "umask #{resource[:umask]};"
end
```
This would be the same as the local_lib parameter code (and that doesn't have a default either).
```ruby
if resource[:local_lib]
  ll = "-Mlocal::lib=#{resource[:local_lib]}"
end
```

Rubocop prefers the single line style but I decided to leave refactoring the local_lib `if` to a completely separate rubocop/refactor PR to come a bit later. :)
> Favor modifier `if` usage when having a single-line body.